dll/win32/shell32/shlexec.cpp

```diff
@@ -2650,3 +2650,197 @@ HRESULT WINAPI ShellExecCmdLine(
 
     return HRESULT_FROM_WIN32(dwError);
 }
+
+/*************************************************************************
+ *                RealShellExecuteExA (SHELL32.266)
+ */
+EXTERN_C
+HINSTANCE WINAPI
+RealShellExecuteExA(
+    _In_opt_ HWND hwnd,
+    _In_opt_ LPCSTR lpOperation,
+    _In_opt_ LPCSTR lpFile,
+    _In_opt_ LPCSTR lpParameters,
+    _In_opt_ LPCSTR lpDirectory,
+    _In_opt_ LPSTR lpReturn,
+    _In_opt_ LPCSTR lpTitle,
+    _In_opt_ LPVOID lpReserved,
+    _In_ INT nCmdShow,
+    _Out_opt_ HANDLE *lphProcess,
+    _In_ DWORD dwFlags)
+{
+    SHELLEXECUTEINFOA ExecInfo;
+
+    TRACE("(%p, %s, %s, %s, %s, %p, %s, %p, %u, %p, %lu)\n",
+          hwnd, debugstr_a(lpOperation), debugstr_a(lpFile), debugstr_a(lpParameters),
+          debugstr_a(lpDirectory), lpReserved, debugstr_a(lpTitle),
+          lpReserved, nCmdShow, lphProcess, dwFlags);
+
+    ZeroMemory(&ExecInfo, sizeof(ExecInfo));
+    ExecInfo.cbSize = sizeof(ExecInfo);
+    ExecInfo.fMask = SEE_MASK_FLAG_NO_UI | SEE_MASK_UNKNOWN_0x1000;
+    ExecInfo.hwnd = hwnd;
+    ExecInfo.lpVerb = lpOperation;
+    ExecInfo.lpFile = lpFile;
+    ExecInfo.lpParameters = lpParameters;
+    ExecInfo.lpDirectory = lpDirectory;
+    ExecInfo.nShow = (WORD)nCmdShow;
+
+    if (lpReserved)
+    {
+        ExecInfo.fMask |= SEE_MASK_FLAG_SEPVDM;
+        ExecInfo.hInstApp = (HINSTANCE)lpReserved;
+    }
+
+    if (lpTitle)
+    {
+        ExecInfo.fMask |= SEE_MASK_HASTITLENEW;
+        ExecInfo.lpClass = lpTitle;
+    }
+
+    if (dwFlags & 1)
+        ExecInfo.fMask |= SEE_MASK_HASTITLE; /* FIXME: Value mismatch */
```

Comment:
OK from analysing Win2000 progman.exe, here is what I find:
![image](https://github.com/reactos/reactos/assets/1969829/aec558d0-2978-4a00-b3f4-89d90eb56c32)

Then, the function I called "ExecuteProgram" above, sets `dwFlags` to 1 when `bSepVDM` is `TRUE`.
This would indicate that what in current master, the value I originally randomly assigned to `SEE_MASK_HASTITLE`, namely `0x00020000`, would actually correspond to the `SEE_MASK_FLAG_SEPVDM` flag.

In conclusion, it would appear these assignations would be more correct:
```c
#define SEE_MASK_FLAG_SEPVDM    0x00020000
#define SEE_MASK_USE_RESERVED   0x00040000
#define SEE_MASK_HASTITLE       0x00080000
```
where the flag `SEE_MASK_USE_RESERVED` (whose name I've made up) is set when lpReserved is not NULL.

Progman.exe passes for the `lpReserved` parameter a string whose format is: `"dde.%d,hotkey.%d,ntvdm.%d"` (with the formatters replaced by some values). They turn out to be what is parsed also here: https://git.reactos.org/?p=reactos.git;a=blob;f=dll/win32/kernel32/client/console/init.c;hb=561ad3c1d6ab4d5f617e54e03a3e9f0feb46cc70#l235

Reply:
How about Wine?

Reply:
Wine has no such functions:

```txt
#define SEE_MASK_DEFAULT          0x00000000
#define SEE_MASK_CLASSNAME        0x00000001
#define SEE_MASK_CLASSKEY         0x00000003
#define SEE_MASK_IDLIST           0x00000004
#define SEE_MASK_INVOKEIDLIST     0x0000000c
#define SEE_MASK_ICON             0x00000010
#define SEE_MASK_HOTKEY           0x00000020
#define SEE_MASK_NOCLOSEPROCESS   0x00000040
#define SEE_MASK_CONNECTNETDRV    0x00000080
#define SEE_MASK_NOASYNC          0x00000100
#define SEE_MASK_FLAG_DDEWAIT     SEE_MASK_NOASYNC
#define SEE_MASK_DOENVSUBST       0x00000200
#define SEE_MASK_FLAG_NO_UI       0x00000400
#define SEE_MASK_UNICODE          0x00004000
#define SEE_MASK_NO_CONSOLE       0x00008000
#define SEE_MASK_ASYNCOK          0x00100000
#define SEE_MASK_HMONITOR         0x00200000
#define SEE_MASK_NOZONECHECKS     0x00800000
#define SEE_MASK_NOQUERYCLASSSTORE 0x01000000
#define SEE_MASK_WAITFORINPUTIDLE 0x02000000
#define SEE_MASK_FLAG_LOG_USAGE   0x04000000
```

Reply:
Yes but, since when are we caring about Wine when it comes to the shell? We also have this "undocumented" flags thing, so it's here (in the place that you originally modified) where they have to be added.

Reply:
I see.